### ISSUE TITLE
Enable cross account resource lookups and actions

### ISSUE THREAD

**Issue by FireballDWF** [capitalone/cloud-custodian#1415]
**Desired behavior:** 
Allow the specification of execution-options: assume_role to use account_id_from_event such as 
          assume_role: arn:aws:iam::{{account_id_from_event}}:role/target-role
so that an inbound event from a cross account delivery via https://aws.amazon.com/blogs/aws/new-cross-account-delivery-of-cloudwatch-events/ uses the event's "account" value are the variable seen above as {{account_id_from_event}}.

This role would then be used for the Describe of the resource and the terminate and/or mark actions.

**Benefits:** 
Enables multi-account scenarios to have one lambda to handle a policy rather a lambda for each account, thus reduce lambda sprawl management.

**Impact of not doing this:** 
Events delivered from other accounts, Custodian fails with the following error:
event ids not resolved: [u'i-0af1e58a1bd481072'] error:An error occurred (InvalidInstanceID.NotFound) when calling the DescribeInstances operation: The instance ID 'i-0af1e58a1bd481072' does not exist

**Comment by kapilt** [comment]
i assume this is resolved with the merged pull request in #1416

**Comment by cplee** [comment]
@FireballDWF did this get implemented as described in your initial comment?  I have a similar use case and i'm hoping to assume cross-account roles based on the account_id in the cloudwatch event.  Is this doable with a single lambda deployment or do I need multiple deployments, one for each account?

**Comment by cplee** [comment]
from @kapilt: 
>  i think the primary reason i never bothered with that, is that it still needs per policy infrastructure in the member account, and separating it out the policy provisioned resources across accounts seemed a bit unwieldy.
>
> ie. you still need the cwe rule and target per policy in each account and then deploy out the lambda into central per policy, any policy across the accounts has to do a delta on the lambda func in the central account. from a resiliency or even metrics collection perspective its a bit unwieldy for unclear benefit. i'd like to get a bit clearer on the value chain, please comment on #1415

My goal is to simplify the deploy process of rule changes.  Currently, if I have 20 member accounts, I must assume a role in each of the accounts and run custodian to deploy the policy lambda functions in those accounts.  

I agree that there is still infrastructure in the member account, but it doesn't have to be a one-to-one between policy and member account infrastructure.  I'm using the CloudFormation template below to configure the member account with a single CW rule/target that would send all ec2 and s3 events to the central account.  This rule would only ever need to change if I needed to start writing policies for a new resource type.  I can deploy the following CFN via a stack set across all member accounts once and then I can change my policies in my central account as often as needed without requiring any updates on the member accounts.

```
  EventForwarder:
    Type: AWS::Events::Rule
    Properties:
      Name: cloud-custodian-forwarder
      EventPattern:
        source:
          - aws.ec2
          - aws.s3
      State: ENABLED
      Targets:
        - Id: ToolsAccountEventBus
          Arn: !Sub arn:aws:events:${ToolsRegion}:${ToolsAccount}:event-bus/default
```

**Comment by kapilt** [comment]
implemented in #2005
